orttraining/orttraining/test/gradient/gradient_ops_test.cc

```diff
@@ -185,6 +185,27 @@ void TestBroadcastableBinaryOpGrad(const std::string& op_type,
     gradient_checker.ComputeGradientError(op_def, {A_info, B_info}, {Y_info}, &max_error);
     EXPECT_IS_TINY(max_error);
   }
+
+  // symbolic broadcast
+  // shape(A) = (4, 2, 1, "seq(3)"), shape(B) = (4, 2, 1, 1), ==> shape(result) = (4, 2, 1, 3)
+  {
+    TensorInfo A_info{{4, 2, 1, 3}, true, transformer, DataTypeImpl::GetTensorType<float>(), {"4", "2", "1", "seq"}};
```

Comment:
>"4", "2", "1" [](start = 94, length = 13)

maybe better to keep dim_value with empty dim_param? so {"", "", "", "seq"}

Reply:
right. This is from my old code when I thought the dim_param value can be different than actual input shape, so I was using the string to distinguish the tensor proto shape v.s. the actual shape.. Thanks for pointing out. I'll update it.

---
In reply to: [418774262](https://github.com/microsoft/onnxruntime/pull/3743#discussion_r418774262) [](ancestors = 418774262)

Reply:
Yeah, there is no functional difference between using the string value v.s. using the actual shape value. I just thought it would be easier and more straight forward for reader to figure out the tensor shape. At least I found it useful for me when I have a small monitor -- I don't need to scroll left and right in two places to know what the tensor shape is. 

---
In reply to: [418774262](https://github.com/microsoft/onnxruntime/pull/3743#discussion_r418774262) [](ancestors = 418774262)